docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_log_file_integrity.mdx

```diff
@@ -23,6 +23,217 @@ HIPAA, CISAWS, CBP, SOC2, NIST, GDPR
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all trails: Use the following AWS CLI command to list all the trails in your AWS account:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+   This command will return a JSON output with details of all the trails.
+
+3. Check File Integrity Validation Feature: For each trail, check if the `LogFileValidationEnabled` attribute is set to `true`. This attribute indicates whether the file integrity validation feature is enabled for the trail. You can do this by running the following command:
+
+   ```
+   aws cloudtrail get-trail-status --name TrailName
+   ```
+   Replace `TrailName` with the name of your trail. If the `LogFileValidationEnabled` attribute is set to `true`, the file integrity validation feature is enabled for the trail.
+
+4. Repeat the process: Repeat step 3 for each trail in your AWS account. If any trail has the `LogFileValidationEnabled` attribute set to `false`, it means the file integrity validation feature is not enabled for that trail, indicating a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. This can be done using pip:
+
+   ```bash
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and set up AWS credentials: In your Python script, you need to import the Boto3 library and set up your AWS credentials (Access Key ID and Secret Access Key). You can do this using the AWS CLI or by manually setting them in your script:
+
+   ```python
+   import boto3
+
+   # Set AWS credentials
+   aws_access_key_id = 'YOUR_ACCESS_KEY'
+   aws_secret_access_key = 'YOUR_SECRET_KEY'
+   region_name = 'us-west-2' # or your preferred region
+
+   # Create a session using your credentials
+   session = boto3.Session(
+       aws_access_key_id=aws_access_key_id,
+       aws_secret_access_key=aws_secret_access_key,
+       region_name=region_name
+   )
+   ```
+
+3. Create a CloudTrail client and list trails: Using the session you created, you can create a CloudTrail client and list all trails in your account:
+
+   ```python
+   # Create CloudTrail client
+   cloudtrail = session.client('cloudtrail')
+
+   # List all trails
+   response = cloudtrail.describe_trails()
+
+   for trail in response['trailList']:
+       print(trail['Name'])
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>